src/scaling/scaling.jl

```diff
@@ -0,0 +1,87 @@
+type ScaledInterpolation{T,N,ITPT,IT,GT,RT} <: AbstractInterpolationWrapper{T,N,ITPT,IT,GT}
+    itp::ITPT
+    ranges::RT
+end
+ScaledInterpolation{T,ITPT,IT,GT,RT}(::Type{T}, N, itp::ITPT, ::Type{IT}, ::Type{GT}, ranges::RT) =
+    ScaledInterpolation{T,N,ITPT,IT,GT,RT}(itp, ranges)
+@doc """
+`scale(itp, xs, ys, ...)` scales an existing interpolation object to allow for indexing using other coordinate axes than unit ranges, by wrapping the interpolation object and transforming the indices from the provided axes onto unit ranges upon indexing.
```

Comment:
Yes - I agree that `interpolate(knots, data, Gridded)` is a better interface, as is probably `scale(knots, itp)` if that was the only thing we did (and the fact that both MATLAB, SciPy and, I assume, a whole bunch of others do it that way is fuel on the fire).

In the context of composition of various "add-ons", though, I think keeping the API as close to `do_extra_stuff(itp, args...)` has a lot of value. `scale` and `extrapolate` are both examples of `do_extra_stuff` in this context. If we don't adhere to `(itp, args...)` for all of them, it will be a lot more difficult for users of *this* library to know (without resorting to docs) in which order the arguments should come.